internal/isoeditor/rhcos.go

```diff
@@ -541,3 +546,49 @@ func ParseOffsetInfo(headerBytes []byte) (*OffsetInfo, error) {
 	}
 	return offsetInfo, nil
 }
+
+func EmbedIgnition(inputISOPath, outputISOPath, ignitionConfig string) error {
+	coreosIgnitionHeader := make([]byte, ignitionHeaderSize)
+
+	inputISO, err := os.Open(inputISOPath)
+	if err != nil {
+		return err
+	}
+	defer inputISO.Close()
+
+	if _, err = inputISO.ReadAt(coreosIgnitionHeader, isoSystemAreaSize-ignitionHeaderSize); err != nil {
+		return err
+	}
+	ignitionOffsetInfo, err := GetIgnitionArea(coreosIgnitionHeader)
+	if err != nil {
+		return err
+	}
+
+	cpio, err := IgnitionImageArchive(ignitionConfig)
+	if err != nil {
+		return err
+	}
+
+	if len(cpio) > int(ignitionOffsetInfo.Length) {
+		return errors.Errorf("Compressed Ignition config is too large: %v > %v", len(cpio), int(ignitionOffsetInfo.Length))
+	}
+
+	resultISO, err := os.Create(outputISOPath)
+	if err != nil {
+		return err
+	}
+	defer resultISO.Close()
+
+	if _, err = io.Copy(resultISO, inputISO); err != nil {
+		return err
+	}
+
+	// clear out the embed area
+	embedArea := make([]byte, ignitionOffsetInfo.Length)
```

Comment:
How big is this?
Is it big enough for us to want to do something like `WriteAt` with a reader from `/dev/zero` or something?

Reply:
It's not clear to me how to know for certain. I "copied" this from https://github.com/coreos/coreos-installer/blob/master/src/live.rs#L565-L567 without much thought to be honest.